lib/main.js

```diff
@@ -385,54 +392,192 @@ function createPipelinedRequest(cb){
-        _options.headers['method']=method;
-    }else{
-        _options=options;
+        _options.headers['host'] = host;
+        _options.headers['port'] = port;
+        _options.headers['path'] = path;
+        _options.headers['method'] = method;
+    } else {
+        _options = options;
+    }
+//--------------------------Anonymizating Request----------------------------------------------------------------------------------
+
+if (options.body && anonymizationEnabled) {
+    const contentType = options.headers && options.headers['Content-Type'] ? options.headers['Content-Type'] : '';
+    console.log('Content-Type:', contentType); 
+    if (options.method === 'POST' || options.method === 'PUT' || options.method === 'PATCH') {
+        switch (true) {
+            //urlencoded
+            case contentType.includes('application/x-www-form-urlencoded'):
+                options.body = anonymizer.anonymizeUrlEncoded(options.body);
+                console.log('Anonymized URL-encoded(Request):', options.body);
+                break;
+            //json
+            case contentType.includes('application/json'):
+                options.body = JSON.stringify(anonymizer.anonymizeObject(JSON.parse(options.body)));
+                console.log('Anonymized JSON(Request):', options.body);
+                break;
+            //form-date
+            case contentType.includes('multipart/form-data'):
+                anonymizer.anonymizeFormData(options.body, (err, anonymizedData) => {
+                    if (!err) {
+                        options.body = anonymizedData;
+                        console.log('Anonymized Multipart Form-data(Request):', options.body);
+                    } else {
+                        console.error('Anonymization error(Request):', err);
+                    }
+                });
+                break;
+            //default set as string
+            default:
+                options.body = anonymizer.anonymizeText(options.body);
+                console.log('Anonymized Other Type(Request):', options.body);
+        }
+    } else if (options.method === 'GET' || options.method === 'DELETE') {
+        anonymizer.anonymizeRequest(options);
+        console.log(`${options.method} request:`, options.body);
+    } else {
+        anonymizer.anonymizeRequest(options);
+        console.log('Anonymized Other HTTP Method(Request):', options.body);
     }
+}
+
+
+//------------------------------------------------------------------------------------------------------------
+    
+
     let protocol;
     if (options.port === "443" && !isThrottled()) {
         _options['agent'] = gh.keepAliveAgentHttps;
         protocol = https;
-    }
-    else {
+    } else {
         _options['agent'] = gh.keepAliveAgentHttp;
         protocol = http;
     }
+
+
+    //--------------------------Anonymizating Response----------------------------------------------------------------------------------
```

Comment:
All the code added here must be part of anonymizer.js. You have to rework this part, and keep the initial implementation of the main.js intact with just the call to anonymizer function that would anonymize the response.

I would expect something like this:

Existing code with minor intervention:
```
var req = protocol.request(_options, function (response) {
        response.on('end', function () {
            incrCounter(response.statusCode);
        });

        if(cb){
            cb(!anonymizer.isOn ? response : anonymizer.manipulateResponse(response));
        }
        response.resume();
    });
```

You can modify(anonymize) the response data before sending it to the client, so you do not break any existing usages of the existing library.

Reply:
looks good! I will change it now